src/Storages/ExternalStream/Kafka/KafkaSource.h

```diff
@@ -76,6 +82,24 @@ class KafkaSource final : public ISource
 
     UInt32 record_consume_batch_count = 1000;
     Int32 record_consume_timeout = 100;
+
+    /// For checkpoint
+    /// FIXME, switch to llvm-15
+    std::atomic<CheckpointContext *> ckpt_ctx;
+    struct State
+    {
+        void serialize(WriteBuffer & wb) const;
+        void deserialize(VersionType version, ReadBuffer & rb);
+
+        static constexpr VersionType VERSION = 0; /// Current State Version
+
+        /// For VERSION-0
```

Comment:
It would be greate if you could elaborate on what `For VERSION-0` actually means here.

Reply:
```c++
/// NOTE: Keeping all data structure of each version.
/// If there are multiple versions in the future, we can achieve version compatibility more clearly
/* For exmpale:
    /// VERSION-1:
    int data_1;
    int data_3;

    /// VERSION-0
    // int data_1;
    // string data_2;
*/
```